addons/website_quote/models/sale_order.py

```diff
@@ -66,7 +66,7 @@ def _get_default_template(self):
     quote_viewed = fields.Boolean('Quotation Viewed')
     require_payment = fields.Selection([
         (0, 'Online Signature'),
-        (1, 'Online Payment')], default=0, string='Confirmation Mode',
+        (1, 'Online Payment')], default=lambda self: 1 if self.env['ir.config_parameter'].sudo().get_param('sale.sale_portal_confirmation_options', default='none') == 'pay' else 0, string='Confirmation Mode',
```

Comment:
I would move it into a real default method. Too long lambda are hard to manipulate and understand.